### ISSUE TITLE
Crash on start of jabberdctl

### ISSUE THREAD

**Issue by bluemle** [processone/ejabberd#4241]
## Environment

- ejabberd version: 24.06-1
- Erlang version: Erlang (SMP,ASYNC_THREADS) (BEAM) emulator version 15.0
- OS: Linux (Arch)
- Installed from: distro package

## Bug description

I get the following error when trying to start jabberdctl as non-root user:

(no logger present) unexpected logger message: {log,error,"Error in process ~p with exit value:~n~p~n",[<0.43.0>,{badarg,[{persistent_term,get,[code_server],[{error_info,#{module=>erl_erts_errors}}]},{code_server,get_mode,0,[{file,"code_server.erl"},{line,75}]},{code,ensure_loaded,1,[{file,"code.erl"},{line,558}]},{error_handler,undefined_function,3,[{file,"error_handler.erl"},{line,84}]},{logger_backend,call_handlers,3,[{file,"logger_backend.erl"},{line,51}]},{proc_lib,exit_p,3,[{file,"proc_lib.erl"},{line,350}]}]}],#{error_logger=>#{emulator=>true,tag=>error},pid=><0.43.0>,time=>1719877673381543,gl=><0.0.0>}}
Kernel pid terminated (logger) ({badarg,[{persistent_term,get,[code_server],[{error_info,#{module=>erl_erts_errors}}]},{code_server,get_mode,0,[{file,"code_server.erl"},{line,75}]},{code,ensure_loaded,1,[{file,"code.erl"},{line,558}]},{error_handler,undefined_function,3,[{file,"error_handler.erl"},{line,84}]},{logger_backend,call_handlers,3,[{file,"logger_backend.erl"},{line,51}]},{proc_lib,exit_p,3,[{file,"proc_lib.erl"},{line,350}]}]})

Does anyone have an idea what causes this error? It does not occur if I run jabberdctl as root.

Thank you

**Comment by licaon-kter** [comment]
What if you start it as its user, that you've setup, like `sudo -u ejabberd ejabberdctl` ?

**Comment by bluemle** [comment]
Yes, I tried this as well but getting the same error unfortunately.
`sudo -u jabber ejabberdctl`

**Comment by badlop** [comment]
> OS: Linux (Arch) 
> Installed from: distro package
> It does not occur if I run jabberdctl as root.

Do you mean that you installed using the Arch package? Maybe it's a problem in that package? You could try installing its latest version. Or maybe that package is designed to be ran bu root?

In the meantime, if you find no other solution, you can of course compile ejabberd yourself, or use a binary installer which should support running by a regular user.

**Comment by dominicletz** [comment]
This seems to be a permission issue on the `pwd` inside erlang. These workarounds work for me:

1. `sudo -i -u ejabberd ejabberdct` - The `-i` option creates a login shell and changes the `pwd` to the users home directory before launching the cmd.
2. Update the service start script to first go to the running users home directory
